changelog/5029.feature.rst

```diff
@@ -0,0 +1,5 @@
+Add ``one_of`` to component. Should be used inside ``requires``.
+
+``one_of`` allows you to define multiple attributes of a message for a component.
+At least one of those attributes needs to be present.
+Otherwise, the component cannot be trained and an error is raised.
```

Comment:
My inspiration was this: https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/

in which case I guess this use case would actually be `any_of` if at least one but possibly multiple must be present. That makes me think -- can we make use of the python native `any()` function somehow?

Reply:
Will rename it to `any_of`. I don't think we can use the builtin function `any`, at least I don't see how.